streams/src/main/java/org/apache/kafka/streams/state/internals/CachingSessionStore.java

```diff
@@ -232,9 +232,15 @@ public void flush() {
     }
 
     public void close() {
-        flush();
-        cache.close(cacheName);
-        super.close();
+        try {
+            flush();
+        } finally {
+            try {
+                super.close();
+            } finally {
+                cache.close(cacheName);
```

Comment:
Why did you change the order of `super.close()` vs `cache.close()` ?

Reply:
Looking at the implementation, I don't think the order matters. But IIRC, it's dubious to throw exceptions in a `finally` block because it creates ambiguity about which exception should ultimately get thrown.

Perhaps we could instead just capture the exceptions like:
```java
final RuntimeException flushException = capture(this::flush);
final RuntimeException cacheException = capture(() -> cache.close(cacheName));
final RuntimeException superCloseException = capture(wrapped::close);
if (flushException == null && cacheException == null && superCloseException == null) {
  return;
else {
  final RuntimeException toThrow = new RuntimeException("Caught an exception while closing " + storeName);
  if (flushException != null) {
    toThrow.addSuppressed(flushException);
  }
  ...
}
```

WDTY?

Reply:
> because it creates ambiguity 

AFAIK, it's not ambiguous: a later thrown exception would "overwrite" the former.